runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1556,65 +1648,35 @@ TR::Node *TR_VectorAPIExpansion::fromBitsCoercedIntrinsicHandler(TR_VectorAPIExp
    }
 
 TR::Node *TR_VectorAPIExpansion::compareIntrinsicHandler(TR_VectorAPIExpansion *opt, TR::TreeTop *treeTop, TR::Node *node,
-                                                       TR::DataType elementType, vec_sz_t vectorLength, handlerMode mode)
+                                                         TR::DataType elementType, TR::VectorLength vectorLength, int32_t numLanes,
+                                                         handlerMode mode)
    {
-   return NULL; // the folowing is a prototype cod until Mask/Shuffle design
-
-   TR::Compilation *comp = opt->comp();
-   TR::Node *opcodeNode = node->getFirstChild();
-
-   if (!opcodeNode->getOpCode().isLoadConst())
-      {
-      if (opt->_trace) traceMsg(comp, "Unknown opcode in node %p\n", node);
-      return NULL;
-      }
-
-   int32_t vectorAPIOpcode = opcodeNode->get32bitIntegralValue();
-   TR::DataType opType = elementType;
+   return naryIntrinsicHandler(opt, treeTop, node, elementType, vectorLength, numLanes, mode, 2, Compare);
+   }
 
-   // Byte and Short are promoted after being loaded from array
-   // and all operations should be done in Int
-   if (elementType == TR::Int8 || elementType == TR::Int16)
-      opType = TR::Int32;
+TR::ILOpCodes TR_VectorAPIExpansion::ILOpcodeFromVectorAPIOpcode(int32_t vectorAPIOpCode, TR::DataType elementType,
+                                                                 TR::VectorLength vectorLength, vapiOpCodeType opCodeType, bool withMask)
+   {
+   // TODO: support more scalarization
 
-   TR::ILOpCodes scalarOpCode = ILOpcodeFromVectorAPIOpcode(vectorAPIOpcode, opType, 0, Compare);
-   TR::ILOpCodes vectorOpCode = ILOpcodeFromVectorAPIOpcode(vectorAPIOpcode, opType, vectorLength, Compare);
+   bool scalar = (vectorLength == TR::NoVectorLength);
+   TR::DataType vectorType = scalar ? TR::NoType : TR::DataType::createVectorType(elementType.getDataType(), vectorLength);
 
-   if (mode == checkScalarization)
+   if (opCodeType == Compare && withMask)
```

Comment:
suggestion for better readability:  make your intention clear ---  `((opCodeType == Compare) && withMask)`